src/main/java/com/booking/superride/service/RideService.java

```diff
@@ -0,0 +1,120 @@
+package com.booking.superride.service;
+
+import com.booking.superride.domain.RideBookingRequest;
+import com.booking.superride.domain.RideDetailsResponse;
+import com.booking.superride.entity.RideDetails;
+import com.booking.superride.entity.TaxiDetails;
+import com.booking.superride.entity.projections.AvailableTaxiFareDetails;
+import com.booking.superride.entity.projections.OnTripTaxiDetails;
+import com.booking.superride.mapper.RideDetailsMapper;
+import com.booking.superride.repository.RideRepository;
+import com.booking.superride.repository.TaxiRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Objects;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class RideService {
+
+    private final RideRepository rideRepository;
+    private final TaxiRepository taxiRepository;
+    private final RideDetailsMapper rideDetailsMapper;
+
+    public RideDetailsResponse bookRide(RideBookingRequest request) {

```

Comment:
Play with transactional boundry, ideally all Read operations should run with @Transactional(readonly=true)